skyportal/models.py

```diff
@@ -2146,16 +2152,132 @@ def send_user_invite_email(mapper, connection, target):
         html_content=(
             f'{cfg["invitations.email_body_preamble"]}<br /><br />'
             f'Please click <a href="{link_location}">here</a> to join.'
         ),
     )
-    sg = SendGridAPIClient(cfg["invitations.sendgrid_api_key"])
+    sg = SendGridAPIClient(cfg["twilio.sendgrid_api_key"])
     sg.send(message)
 
 
+class SourceAlert(Base):
+    groups = relationship(
+        "Group",
+        secondary="group_alerts",
+        cascade="save-update, merge, refresh-expire, expunge",
+        passive_deletes=True,
+    )
+    sent_by = relationship(
+        "User",
+        secondary="user_alerts",
+        cascade="save-update, merge, refresh-expire, expunge",
+        passive_deletes=True,
+        uselist=False,
+    )
+    source_id = sa.Column(
+        sa.ForeignKey('objs.id', ondelete='CASCADE'),
+        nullable=False,
+        index=True,
+        doc="ID of the target Obj.",
+    )
+    source = relationship('Obj', back_populates='source_alerts', doc='The target Obj.')
+
+    additional_notes = sa.Column(sa.String(), nullable=True)
+    level = sa.Column(sa.String(), nullable=False)
+
+
+GroupSourceAlert = join_model('group_alerts', Group, SourceAlert)
+UserSourceAlert = join_model("user_alerts", User, SourceAlert)
+
+
+@event.listens_for(SourceAlert, 'after_insert')
+def send_source_alert(mapper, connection, target):
+    _, cfg = load_env()
+    ports_to_ignore = {True: 443, False: 80}  # True/False <-> server.ssl=True/False
+    app_base_url = (
+        f"{'https' if cfg['server.ssl'] else 'http'}:"
+        f"//{cfg['server.host']}"
+        + (
+            f":{cfg['server.port']}"
+            if (
+                cfg["server.port"] is not None
+                and cfg["server.port"] != ports_to_ignore[cfg["server.ssl"]]
+            )
+            else ""
+        )
+    )
+
+    link_location = f'{app_base_url}/source/{target.source_id}'
+    if target.sent_by.first_name is not None and target.sent_by.last_name is not None:
+        sent_by_name = f'{target.sent_by.first_name} {target.sent_by.last_name}'
+    else:
+        sent_by_name = target.sent_by.username
+
+    group_ids = map(lambda group: group.id, target.groups)
+    groups = DBSession().query(Group).filter(Group.id.in_(group_ids)).all()
+
+    target_users = set()
+    for group in groups:
+        # Use a set to get unique iterable of users
+        target_users.update(group.users)
+
+    source = DBSession().query(Obj).filter(Obj.id == target.source_id).one()
+    source_info = ""
+    if source.ra is not None:
+        source_info += f'RA={source.ra} '
+    if source.dec is not None:
+        source_info += f'Dec={source.dec}'
+    source_info = source_info.strip()
+
+    # Send SMS messages to opted-in users if desired
+    if target.level == "hard":
+        message_text = (
+            f'{cfg["app.title"]}: {sent_by_name} would like to call your immediate'
+            + f' attention to a source at {link_location} ({source_info}).'
```

Comment:
```suggestion
            f' attention to a source at {link_location} ({source_info}).'
```
the parentheses takes care of that